crates/sui-package-resolver/src/lib.rs

```diff
@@ -356,6 +359,96 @@ impl<S: PackageStore> Resolver<S> {
 
         Ok(sigs)
     }
+
+    /// Attempts to infer the type layouts for pure inputs to the programmable transaction.
+    ///
+    /// The returned vector contains an element for each input to `tx`. Elements corresponding to
+    /// pure inputs that are used as arguments to transaction commands will contain `Some(layout)`.
+    /// Elements for other inputs (non-pure inputs, and unused pure inputs) will be `None`.
+    ///
+    /// Layout resolution can fail if a type/module/package doesn't exist, if layout resolution hits
+    /// a limit, or if a pure input is somehow used in multiple conflicting occasions (with
+    /// different types).
+    pub async fn pure_input_layouts(
```

Comment:
There are a couple of calls to `await` in here inside loops (calls to `function_parameters` and to `type_layouts`). I chose not to run this work concurrently because the majority of its time would be spent waiting for the lock on the package cache, so I expected the communication/locking overhead to be higher than the potential concurrency win.